include/samurai/mesh_config.hpp

```diff
@@ -0,0 +1,461 @@
+// Copyright 2018-2025 the samurai's authors
+// SPDX-License-Identifier:  BSD-3-Clause
+
+#pragma once
+
+#include "arguments.hpp"
+#include "cell_array.hpp"
+#include "samurai_config.hpp"
+
+#include <array>
+
+namespace samurai
+{
+
+    template <std::size_t dim_,
+              int prediction_stencil_radius_    = default_config::prediction_stencil_radius,
+              std::size_t max_refinement_level_ = default_config::max_level,
+              class interval_t_                 = default_config::interval_t>
+    class mesh_config
+    {
+      public:
+
+        static constexpr std::size_t dim                  = dim_;
+        static constexpr int prediction_stencil_radius    = prediction_stencil_radius_;
+        static constexpr std::size_t max_refinement_level = max_refinement_level_;
+
+        using interval_t = interval_t_;
+
+        using cell_t   = Cell<dim, interval_t>;
+        using cl_type  = CellList<dim, interval_t, max_refinement_level>;
+        using lcl_type = typename cl_type::lcl_type;
+
+        using ca_type  = CellArray<dim, interval_t, max_refinement_level>;
+        using lca_type = typename ca_type::lca_type;
```

Comment:
Do you really use these types ? They are declared in `Mesh_base`. If not, could you remove these lines ?

Reply:
I remove all types in `mesh_config`. I was thinking of including all the types that `Mesh_base` needs, but I changed my mind.